Added Capacities for US_PJM, US_NY

HansHyde opened the pull request:
Created US_PJM in countrytopos.js for PJM
(approximated for now until custom geometries capable)

NY data from wikipedia
PJM data fromhttp://www.pjm.com/-/media/markets-ops/ops-analysis/capacity-by-fuel-type-2017.ashx?la=en

corradio commented on config/zones.json:
```diff
@@ -1524,12 +1524,49 @@
+      "demand response": 9112,
+      "energy efficiency": 1790,
+      "external": 196,
+      "unknown": 18,
```
Unfortunately we only handle a given set of keys (so you have to remove/merge biogas, other renewable, demand response, energy efficiency and external)

corradio replied:
This comment hasn't been addressed yet.

corradio commented on config/zones.json:
```diff
@@ -1524,12 +1524,49 @@
+      "oil": 8260,
+      "solar": 250,
+      "wind": 1010,
+      "biomass": 50
```
👍

HansHyde replied:
this is for NY - it was accepted into the master?

corradio replied:
not yet I think

corradio commented on web/app/countrytopos.js:
```diff
@@ -413,25 +413,26 @@ exports.addCountryTopos = function(countries) {
-    countries['US-PA'] = getState('USA', 'US.PA')
+    // countries['US-PA'] = getState('USA', 'US.PA')
     // countries['US-RI'] = getState('USA', 'US.RI')
+    countries['US-PJM'] = getStates('USA', ['US.DE', 'US.KY', 'US.MD', 'US.NJ', 'US.OH', 'US.PA', 'US.VA', 'US.WV'])
```
Looks good. Are you able to start the docker containers and submit a screenshot? Then I don't have to double check myself :)

HansHyde replied:
Looking into it now.. I have Docker installed... need to youtube the rest. :D

corradio replied:
let us know on Slack if you have any issues!

corradio commented:
Also, will need to update the README with the sources for the data updates.

corradio commented on config/zones.json:
```diff
@@ -1524,12 +1524,49 @@
+      "unknown": 18,
+    },
+    "flag_file_name": "us.png",
+    "parsers": {
```
this section should be omitted for now as the parser is not written yet.

corradio commented on README.md:
```diff
@@ -114,6 +114,7 @@ Real-time electricity data is obtained using [parsers](https://github.com/tmrowc
 - United States (California): [CAISO](http://www.caiso.com/Pages/default.aspx)
 - United States (New England): [NEISO](https://www.iso-ne.com/isoexpress/)
 - United States (New York): [NYISO](http://www.nyiso.com/public/markets_operations/market_data/graphs/index.jsp)
+- United States (PJM): [PJM](http://www.pjm.com/markets-and-operations.aspx)
```
Remove for now as the parser hasn't been written.

corradio commented on README.md:
```diff
@@ -208,6 +209,7 @@ Production capacities are centralized in the [zones.json](https://github.com/tmr
 - United States of America:
   - Federal [EIA](https://www.eia.gov/electricity/data.cfm#gencapacity)
   - States [EIA](https://www.eia.gov/electricity/data/state/)
+  - PJM [PJM](http://www.pjm.com/-/media/markets-ops/ops-analysis/capacity-by-fuel-type-2017.ashx?la=en)
```
👍

corradio commented on config/zones.json:
```diff
@@ -1524,12 +1524,48 @@
+      "wind": 1114,
+      "battery discharge": null,
+      "other": 9112,
+      "unknown": 700,
```
Invalid keys that should be removed / merged:
- battery discharge
- other (should be merged with unknown)

Also, values that are 0 should be set to 0 instead of null. Null means that the value is unknown.
This could apply to geothermal / solar / hydro storage / biomass.

HansHyde replied:
Copy, the capacity value is not unknown, it is just not being distinctly reported by PJM - solar & biomass to the best of my understanding is reported under PJM's "other renewables" 

"battery discharge" & "other" are in the co2eq_parameters.js  - my bad for including.

This is what I'm attempting to capture/merge between EM & PJM 

![pjm_elelctricity map - merge cat realtime data](https://user-images.githubusercontent.com/33690988/34340088-bb0a98c4-e97d-11e7-8d4b-c51d0c5eec82.png)

systemcatch replied:
Couple of formatting issues that are making the tests fail, line 1543 should have a comma between the 2 contributor names, line 1559 does not need a comma at the end.

corradio commented on config/zones.json:
```diff
@@ -1524,12 +1524,48 @@
+      "unknown": 700,
+    },
+    "flag_file_name": "us.png",
+    "parsers": {
```
Should be removed as we don't have the parsers ready yet. This will crash the backend.

HansHyde replied:
copy, done - all lines related to "flag_file_name" & "parsers"

HansHyde commented:
do I create a new pull request to submit the changes for PJM?  The edits/corrections are in my master, not sure how to push them into the EM/master with this open commit...

Here are the corrections (as submitted)...

![pjm capacities - as built](https://user-images.githubusercontent.com/33690988/34339856-70489726-e97a-11e7-9796-44d02c92fc09.png)

Here are notes on them... as the zones.json doesn't appear to allow any backend notes regarding the code (not submitted)...

![pjm capacities - with notes](https://user-images.githubusercontent.com/33690988/34339893-d39c95e8-e97a-11e7-85d7-4f8f21c4a9e3.png)

and... can I just remove lines 1563 --> 1566 without issue as per your other message

corradio commented:
Let's keep this pull-request focussed on capacities for both NY and PJM
:+1: (no need to open a new one)
L1564->1566 can indeed just be removed.

On Mon, Dec 25, 2017 at 1:57 PM, Hans Hyde <notifications@github.com> wrote:

> do I create a new pull request to submit the changes for PJM? The
> edits/corrections are in my master, not sure how to push them into the
> EM/master with this open commit...
>
> Here are the corrections (as submitted)...
>
> [image: pjm capacities - as built]
> <https://user-images.githubusercontent.com/33690988/34339856-70489726-e97a-11e7-9796-44d02c92fc09.png>
>
> Here are notes on them... as the zones.json doesn't appear to allow any
> backend notes regarding the code (not submitted)...
>
> [image: pjm capacities - with notes]
> <https://user-images.githubusercontent.com/33690988/34339893-d39c95e8-e97a-11e7-85d7-4f8f21c4a9e3.png>
>
> and... can I just remove lines 1563 --> 1566 without issue as per your
> other message
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tmrowco/electricitymap/pull/942#issuecomment-353867910>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABlEKOLN2b9-LEaOOoEK8hVj8Ym-ZE4rks5tD5udgaJpZM4RLne8>
> .
>

corradio commented on config/zones.json:
```diff
@@ -1524,12 +1524,43 @@
+      "oil": 9148,
+      "solar": null,
+      "wind": 1114,
+      "battery discharge": null,
```
Battery discharge should be removed.

HansHyde replied:
done

HansHyde commented:
> ERROR in ../config/zones.json
> Module build failed: SyntaxError: Unexpected string in JSON at position 34508
>     at JSON.parse (<anonymous>)
>     at Object.module.exports (/home/web/node_modules/json-loader/index.js:4:49)
>  @ ./app/main.js 43:19-53
>  @ multi babel-polyfill ./app/main.js
> 
> ERROR in ./node_modules/json-loader!../config/zones.json
> Module build failed: SyntaxError: Unexpected string in JSON at position 34508
>     at JSON.parse (<anonymous>)
>     at Object.module.exports (/home/web/node_modules/json-loader/index.js:4:49)
>  @ ./app/flags.js 5:12-58
>  @ ./app/main.js
>  @ multi babel-polyfill ./app/main.js

 

Position 34508 is not related to code text... changed it multiple times.  Latest edit put it at start of line for "nuclear"

Removed US-PJM section all together, position 34508 was further down (didn't have character counter yet).  Restarted computer also.

![position 34508 error copy](https://user-images.githubusercontent.com/33690988/34382165-591c4374-eb0d-11e7-811e-e571ea2b271d.png)


![position 34508 error](https://user-images.githubusercontent.com/33690988/34382166-59383804-eb0d-11e7-8212-10d568c4aedd.png)

corradio commented on config/zones.json:
```diff
@@ -1524,12 +1524,37 @@
     },
     "flag_file_name": "us.png",
     "contributors": [
       "https://github.com/systemcatch"
```
Missing comma at end of line.

HansHyde replied:
@corradio I thought that too, but every final line in Capacity like 1536 above ie 1516, 1480, etc. does not have a comma.  I may have messed everything up with my wrong PR requests up top....  "Merge branch 'master' into master 4061f24"

All things equal, I think best for you to close this PR (deny all changes), then I wipe/erase my fork & start fresh.  I can replicate any losses easily enough & I'll be able to see if I can get the mockerserver running from the start. 

Plus I have info ready for PJM, MISO & maybe SWPP.

corradio replied:
@HansHyde I'm talking about the first line.
This is invalid json:
```json
"contributors": [
    "https://github.com/systemcatch"
    "https://github.com/hanshyde"
```

However, this is valid:
```json
"contributors": [
    "https://github.com/systemcatch",
    "https://github.com/hanshyde"
```

HansHyde commented:
Yahooo!  :D  

![yahooo nyiso has capacities](https://user-images.githubusercontent.com/33690988/34386160-9abd7a36-eb27-11e7-92d2-fc7e0e101917.png)

corradio commented on config/zones.json:
```diff
@@ -383,12 +383,21 @@
   },
   "CA-QC": {
     "_comment": "http://www.hydroquebec.com/generation/",
+    "_comment": "http://www.hydroquebec.com/decom",
```
You can't have duplicated keys

HansHyde replied:
removed - no CA_QC.py parser (or in progress) to move to - will save info offline

corradio commented on README.md:
```diff
@@ -57,6 +57,7 @@ Real-time electricity data is obtained using [parsers](https://github.com/tmrowc
 - Canada (Nova Scotia): [Nova Scotia Power](http://www.nspower.ca/en/home/about-us/todayspower.aspx)
 - Canada (Ontario): [IESO](http://www.ieso.ca/power-data)
 - Canada (Prince Edward Island): [Government of PEI](http://www.gov.pe.ca/windenergy/chart.php)
+- Canada (Quebec): [HydroQuebec]http://www.hydroquebec.com/reliability-coordinator/
```
Move this to the capacity section?

HansHyde replied:
I can, but this will be where realtime data comes from (if HQ opens it). See last comment on #166  --> The "Reliability Coordinator" is basically the ISO.

corradio replied:
Let's add this when we actually have realtime data written in the parser. Before, there's no reason to add it (we're being inconsistent else).
For now, let's remove.

corradio commented on README.md:
```diff
@@ -208,6 +209,7 @@ Production capacities are centralized in the [zones.json](https://github.com/tmr
 - United States of America:
   - Federal [EIA](https://www.eia.gov/electricity/data.cfm#gencapacity)
   - States [EIA](https://www.eia.gov/electricity/data/state/)
+  - PJM [PJM](http://www.pjm.com/-/media/markets-ops/ops-analysis/capacity-by-fuel-type-2017.ashx?la=en)
```
Move this to the capacity section?

HansHyde replied:
- added to zones.json US_PJM... Line 1555 "    "_comment": "Installed capacity in PJM RTO effective June 1, 2017 - http://www.pjm.com/-/media/markets-ops/ops-analysis/capacity-by-fuel-type-2017.ashx?la=en"," 
- but README.md line 212 is under line 120 "### Production capacity data sources" so this is also correct, no?

corradio replied:
My bad, this line is correct :)

corradio commented on config/zones.json:
```diff
@@ -383,12 +383,21 @@
+    "flag_file_name": "ca.png",
+    "contributors": [
+      "https://github.com/jarek",
+      "https://github.com/corradio",
```
No need to add me (I'm merely reviewing)

HansHyde replied:
removed

corradio commented on parsers/US_NY.py:
```diff
@@ -8,13 +8,21 @@
-           'Wind': 'wind',
-           'Hydro': 'hydro'
+           'Other Fossil Fuels': 'coal', # NYISO states "Power produced by either coal or oil" - assume coal, oil only power plants only as last resort
+           'Other Renewables': 'biomass', # NYISO states "Power produced by either solar energy, refuse, methane or wood"
```
I'm not sure I'd assume biomass here as solar might be a significant chunk.
Having 253MW during the night only gives us an idea of biomass production, but not of solar production.
We should look around noon, and if numbers are close to 250, then we can exclude solar, although this isn't future proof (when NYISO adds more solar capacity).

HansHyde replied:
Solar is not a significant chunk of anything in NY
Page 42 - http://www.nyiso.com/public/webdocs/media_room/publications_presentations/Power_Trends/Power_Trends/2016-power-trends-FINAL-070516.pdf

Solar is not coming in vast supplies to NY, and it is not produced during the winter.

Can you trust me on this one? - there is not a way to note & link to EM & pthyon files all the resources I am pulling this "well informed information" from.

corradio commented on parsers/US_NY.py:
```diff
@@ -8,13 +8,21 @@
 
 mapping = {
-           'Dual Fuel': 'unknown',
+           'Dual Fuel': 'gas', # NYISO states "Power produced by facilities capable of using natural gas and/or other fuels"
```
👍

corradio commented on parsers/US_NY.py:
```diff
@@ -8,13 +8,21 @@
-           'Other Renewables': 'unknown',
-           'Wind': 'wind',
-           'Hydro': 'hydro'
+           'Other Fossil Fuels': 'coal', # NYISO states "Power produced by either coal or oil" - assume coal, oil only power plants only as last resort
```
Any source to document that oil is only being used as last resort?

HansHyde replied:
it is buried in the 500+ pages State of the NYISO Market - there was a chart showing oil burn went to <5% from 2015 to 2016 of all "dual fuel" capacity plants throughout the respective year.  "as a last resort" might be happening right now due to the cold snap, constrained NG pipelines, not enough NG put into storage before the winter heating season & LNG production increasing.  I will go dig it out, but this is backend information.

corradio commented on parsers/US_NY.py:
```diff
@@ -8,13 +8,21 @@
+           'Other Renewables': 'biomass', # NYISO states "Power produced by either solar energy, refuse, methane or wood"
+                # Based on only 250 MW solar-utility & est 500 MW solar-'behind the meter' (unreportable by NYISO),
+                # and at 0445am NYISO reported 253 MW production --> biomass best assumption
+           'Wind': 'wind', # 2016 Wind capacity is 1754 MW (updated zones.json)
```
Can you remove the comment? (does not add any value and will be outdated).
Also, can you fix the indentation?

HansHyde replied:
- removed wind comment
- put all comments at same indentation, with sub-notes indented further

corradio commented on parsers/US_NY.py:
```diff
@@ -8,13 +8,21 @@
+                # and at 0445am NYISO reported 253 MW production --> biomass best assumption
+           'Wind': 'wind', # 2016 Wind capacity is 1754 MW (updated zones.json)
+           'Hydro': 'hydro' # NYISO states "Power produced by water including pumped storage facilities"
+                # hoping for further data source acquistion to distinguish Hydro from PS
```
can you remove this comment?

HansHyde replied:
-rewrote as information to @systemcatch as the builder of the parser - working on acquiring realtime data from NYPA - NY Power Authority.  They operate 97.7% of all hydro capacity in NY at 3 different locations.

corradio commented on parsers/US_NY.py:
```diff
@@ -8,13 +8,21 @@
+           'Hydro': 'hydro' # NYISO states "Power produced by water including pumped storage facilities"
+                # hoping for further data source acquistion to distinguish Hydro from PS
+
+           # Verification data sources via...
```
Can you fix indentations?

HansHyde replied:
done

corradio commented on config/zones.json:
```diff
@@ -1524,12 +1531,39 @@
     ]
   },
   "US-NY": {
+    "capacity": {
```
If I'm not mistaken we are missing a reference to US-NY capacities in the README.

corradio replied:
Still needs to be addressed

HansHyde replied:
long story short, still working on

HansHyde replied:
it's on its way in commit 271d460 (below)

Sorting how I seem to have painted myself into a corner creating a branch "add-PJM-Parser" 5 days ago. ~hans

corradio commented on config/zones.json:
```diff
@@ -1524,12 +1531,39 @@
+    ]
+  },
+  "US-PJM": {
+    "_comment": "Installed capacity in PJM RTO effective June 1, 2017 - http://www.pjm.com/-/media/markets-ops/ops-analysis/capacity-by-fuel-type-2017.ashx?la=en",
```
No need for this comment as we already have it in the readme (duplicate information is hard to maintain)

corradio commented on parsers/US_NY.py:
```diff
@@ -8,13 +8,20 @@
            'Nuclear': 'nuclear',
-           'Other Fossil Fuels': 'unknown',
-           'Other Renewables': 'unknown',
+           'Other Fossil Fuels': 'coal',    # NYISO states "Power produced by either coal or oil" - assume coal, oil only power plants only as last resort
```
Following up on https://github.com/tmrowco/electricitymap/pull/942#discussion_r159056769.
Let's add just a reference to document this. The 8260MW of installed oil capacity must come from somewhere (right now we assume it to be in "Other Fossil Fuels", so we need to be sure if we exclude it).

HansHyde replied:
Copy, just checked the largest oil only generator in NY - Oswego Harbor @1,804 MW - 2015 & 2016 it had a capacity factor of 0% as per Synapse Energy (Google Earth) that pulls information from US EIA.  As well as a few others Ravenswood & Roseton.  

I think what we have here is poor data on Generation Capacity that I originally pulled from Wikipedia https://en.wikipedia.org/wiki/List_of_power_stations_in_New_York  In reality these are "dual fuel" plants, so I've double counted in capacity.  I'll cleanup, which will drop Oil capacity to around 2,000 MW.

Would it be possible to split a real time data stream within the parser 50/50 into the 2 fuel types "coal" & "oil"?  Then if NYISO reports 1,500 MW under "other fossil fuels" the bars will show 750 MW of coal and 750 MW of oil?  Ultimately we are talking about 3,000 MW of combined coal & oil capacity - each with very low capacity factors over the year, against ~33,000 MW total installed.  The GWHr per year are a handful (less than 1%) compared to overall state generation of ~142,346 GWhr in 2015.

corradio replied:
It's possible but against our operating principles. Our philosophy is to try to do the least amount of assumptions as possible.
If there is an unknown is the data split, let's classify as unknown. This will also trigger visitors to ask questions and push on the data provider to provide better data.

HansHyde replied:
"Other Fossil Fuel" NYISO reporting = coal & oil
- Coal = 1,027 MW
- Oil - only = 2,877 MW
---> but US_NY parser cannot "split" single data streams into two fuel types "coal" & "oil"

"Natural Gas" NYISO reporting = natural gas
- NG - only = 3,898 MW
---> US_NY parser no issues attributing data properly as "gas"

"Dual Fuel" NYISO reporting = generators that can burn oil, gas or both together
- O&G - "Dual  Fuel" = 20,253 MW
---> US_NY parser, which way "gas" or "unknown"?

Data that says the parser registers "dual fuel" as "gas"....
- page 17, Table 2  note "fuel oil" is less than "coal"
- page A-8, 
"_Some generators in New York have dual-fuel capability, allowing them to burn either oil or natural gas. These generators usually burn the most economic fuel, although some may burn oil even when it is more expensive if natural gas is difficult to obtain on short notice or if there is uncertainty about its availability. In addition, New York City and Long Island reliability rules (which are discussed further in Section V.E) sometimes require that certain units burn oil in order to limit the exposure of the electrical grid to possible disruptions in the supply of natural gas. Since most large steam units can burn residual fuel oil (No.6) or natural gas, the effects of natural gas price spikes on power prices are partly mitigated by generators switching to fuel oil._"
- pg A-11, "_Average oil-fired generation fell substantially from 2015, primarily because lower natural gas prices in the first quarter of 2016 made oil production less economic._"
-pg A-180, Table A-11... note columns "Real Time Run Hours" greater than 99% of "Dual Fuel" units run on gas.  DF on Oil mostly zeros.  The few not "0" are in the double digits against year total of 8760 hours.  20 hrs = 0.228% and 100 hrs = 1.142%

Evidence supports listing "dual fuel" as "gas"

NG + Dual Fuel = 24,152 MW capacity under the "Gas" bar for US_NY parser

http://www.nyiso.com/public/webdocs/markets_operations/documents/Studies_and_Reports/Reports/Market_Monitoring_Unit_Reports/2016/NYISO_2016_SOM_Report_5-10-2017.pdf


Side point regarding Hydro & PS within the report.... in other words we are having generation kWh removed from the NYISO report equal to 2x the pumped storage operation... so 2 x "24"gCO2eq/kWh  If all PS is running 1,411 MW x 24g = 33,852 g/kW is "missing".   grrrrr

- pg A-9 "_**Pumped-storage resources in pumping mode are treated as negative generation.** The “Other” category includes methane, refuse, solar, and wood._"


sorry, have to jet but wanted to get this info out there.

corradio commented on parsers/US_NY.py:
```diff
@@ -8,13 +8,20 @@
-           'Other Fossil Fuels': 'unknown',
-           'Other Renewables': 'unknown',
+           'Other Fossil Fuels': 'coal',    # NYISO states "Power produced by either coal or oil" - assume coal, oil only power plants only as last resort
+           'Other Renewables': 'biomass',   # NYISO states "Power produced by either solar energy, refuse, methane or wood"
```
Following up on https://github.com/tmrowco/electricitymap/pull/942#discussion_r159057367:
@HansHyde unfortunately it's not simply a matter of trust as 3rd parties look at our data sources and want to verify for themselves. "Hans Hyde" as a source wouldn't work 😆 
Given the report you linked, it seems there is ~500MW of installed solar capacity at distribution level. That means that we can't exclude the fact that Other Renewables could contain an amount of solar significant compared to biomass.
I can seem to find solid evidence to definitely classify Other Renewables as biomass, so I'd keep it as unknown for now.

HansHyde replied:
Long response short - I'm pulling apart the 2017 NYISO "Gold Book" - It will answer our Generation capacities & fuel types questions no matter what (and whether we like the numbers or not) - data doesn't lie! :D  Spoiler alert - there are only 31.5 MW of installed PV reportable to NYISO, with very little in the pipeline.  "Counting" distributed (behind the meter) PV might be similar to "counting" Demand Response (like the 9,000 MW that PJM was reporting) - if there is no data, does it really exist? ;)

corradio replied:
I'm fine with that. We just need a solid reference to the 31.5 MW number you're reporting and add it the comment.

corradio commented on parsers/US_NY.py:
```diff
@@ -8,13 +8,20 @@
 
 mapping = {
-           'Dual Fuel': 'unknown',
+           'Dual Fuel': 'gas',              # NYISO states "Power produced by facilities capable of using natural gas and/or other fuels"
```
Naive question: does Dual Fuel only mean gas + oil? Could it be gas + biomass? or coal?

HansHyde replied:
for NY's purposes it is only gas or oil.   They can't feed coal or biomass into the units.

In the SE US we get into "dual fuel" coal & biomass.  Usual location is a paper mill, with onsite generation.  They feed coal, tree bark and "black liquor" (sap & tannins), any recovered or unusable byproducts into the boilers.

HansHyde replied:
not self-promotion, but here is [me] from 380ft above a paper factory (from 1 of 3 exhaust stacks) in South Carolina - where they do the above (coal, biomass & black liquor)  https://www.instagram.com/p/bG4AoEtl-b/?taken-by=hansworldtravels

corradio commented on parsers/US_PJM.py:
```diff
@@ -0,0 +1,179 @@
+# The arrow library is used to handle datetimes
```
Are you sure you want the parser to be part of this pull request?
I will only be able to merge once the parser works.

HansHyde replied:
@corradio - I'm working on 4 or 5 parsers/geographies at once.  That US_PJM.py commit went here is my mistake (I really don't know how to use Github - Systemcatch is helping me a bit on Slack, but it is honestly an unfamiliar system/language to me and I need "how to do it" explained in "bubba speak" - another way of saying for a Github idiot [me])

We need "issues" or "Pull Requests" open for each of these areas, whether you prefer to refer to them as Parsers (as the main driver) or "geographies" doesn't matter to me.  So at minimum [again I am an idiot in Github] we can attribute commits accordingly.

For North America (at present time), we need 1 for each of the following...
- Ontario (maybe within Canada is ok as it is substantially finished)
- Quebec
- NE-ISO (New England) - already open
- NYISO (New York) - already open, but a dumping ground for PJM too
- PJM (Penn, NJ, Ohio, etc) - currently being dumped here
- MISO (Michigan, Wisconsin, Indiana, Minnesota, Manitoba (also), others)
- TVA (Tennessee)
- ERCOT (Texas)
- SwPP (Prairie States)

That should suffice for now.  I know that I can commit accordingly (and focused) this way, but without open (whatever they are - issues or PRs) we'll never close them with cross mingled dumping grounds like "add Canada" or "add USA".

If you as admin can back or delete the US_PJM.py commit, that would be great!  I would highly appreciate it! :)

corradio replied:
Unfortunately there's absolutely nothing I can do.
A pull request is a request to the main repository (tmrowco/electricitymap/master) to pull the changes from your fork (hanshyde/electricitymap/master), so I have no power to modify anything.
I suggest that you separate your changes in branches (hanshyde/electricitymap/pjm-capacity etc...) and that you submit a separate pull request for each instead of submitting one from your master branch.
Pragmatically, you can create a branch for the parser, and move it to there instead.

jarek commented on config/zones.json:
```diff
@@ -385,10 +384,17 @@
+      "oil": 130,
+      "wind": 3508,
+      "unknown": 1056,
+      "nuclear": 657
```
IMO this change to Quebec should not be part of a change adding PJM info. There is of course the chain of imports, but following that logic we'd add information for every U.S. zone and the change would be unmanageably big.

Apart from that I would dispute adding nuclear which is permanently shut down.

cclauss commented on parsers/US_PJM.py:
```diff
@@ -0,0 +1,179 @@
+    return data
+
+
+if __name__ == '__main__':
```
print() is a function in Python 3

HansHyde commented:
@corradio & @systemcatch - I have a giant mess on my EM repo I will never be able to sort.  If I delete it (my EM repo), will it allow this PR to be closed out by basically removing all my "open/outstanding/failed" commits from the EM master here?

Any commits I've "added" that would be "lost", has been (or can be) replicated easily enough.

I can just create a new fork to the EM master (current) on my end.

corradio commented:
You can close it yourself actually. But I'll close it now (we can always re-open if needed)

HansHyde commented:
Copy,

and if I scroll down from here, to "Delete this repository", and delete it... it will only impact me?

Then from my text editor (Atom) desktop I can push/pull and all will clear? 

![delete my repository only](https://user-images.githubusercontent.com/33690988/34672925-b4f99f58-f480-11e7-9199-a0f2708f87e4.png)

systemcatch commented:
Yup it won't effect the EM.  When you fork the repo again you will need to set it up again in github desktop.

Key point is that you need to avoid working on your 'master' branch directly, create a new branch for each change you work on, and keep each of those changes focused on one area.
